x-pack/plugins/alerting/common/default_rule_aggregation.ts

```diff
@@ -0,0 +1,183 @@
+      terms: { field: 'alert.attributes.tags', order: { _key: 'asc' }, size: maxTags },
+    },
+    snoozed: {
+      nested: {
+        path: 'alert.attributes.snoozeSchedule',
+      },
+      aggs: {
+        count: {
+          filter: {
+            exists: {
+              field: 'alert.attributes.snoozeSchedule.duration',
+            },
+          },
+        },
+      },
+    },
+  };
+};
+
+export const formatDefaultAggregationResult = (
+  aggregations?: RuleAggregationResult
+): RuleAggregationFormattedResult => {
+  if (!aggregations) {
+    // Return a placeholder with all zeroes
+    const placeholder: RuleAggregationFormattedResult = {
+      ruleExecutionStatus: {},
+      ruleLastRunOutcome: {},
+      ruleEnabledStatus: {
+        enabled: 0,
+        disabled: 0,
+      },
+      ruleMutedStatus: {
+        muted: 0,
+        unmuted: 0,
+      },
+      ruleSnoozedStatus: { snoozed: 0 },
+      ruleTags: [],
+    };
+
+    for (const key of RuleExecutionStatusValues) {
+      placeholder.ruleExecutionStatus[key] = 0;
+    }
+
+    return placeholder;
+  }
+
+  const ruleExecutionStatus = aggregations.status.buckets.map(({ key, doc_count: docCount }) => ({
+    [key]: docCount,
+  }));
+
+  const ruleLastRunOutcome = aggregations.outcome.buckets.map(({ key, doc_count: docCount }) => ({
+    [key]: docCount,
+  }));
+
+  const ret: RuleAggregationFormattedResult = {
+    ruleExecutionStatus: ruleExecutionStatus.reduce(
+      (acc, curr: { [status: string]: number }) => Object.assign(acc, curr),
+      {}
+    ),
+    ruleLastRunOutcome: ruleLastRunOutcome.reduce(
+      (acc, curr: { [status: string]: number }) => Object.assign(acc, curr),
+      {}
+    ),
+    ruleEnabledStatus: {
+      enabled: 0,
+      disabled: 0,
+    },
+    ruleMutedStatus: {
+      muted: 0,
+      unmuted: 0,
+    },
+    ruleSnoozedStatus: { snoozed: 0 },
+    ruleTags: [],
+  };
+
+  // Fill missing keys with zeroes
+  for (const key of RuleExecutionStatusValues) {
+    if (!ret.ruleExecutionStatus.hasOwnProperty(key)) {
+      ret.ruleExecutionStatus[key] = 0;
+    }
+  }
+  for (const key of RuleLastRunOutcomeValues) {
+    if (!ret.ruleLastRunOutcome.hasOwnProperty(key)) {
+      ret.ruleLastRunOutcome[key] = 0;
+    }
+  }
+
+  const enabledBuckets = aggregations.enabled.buckets;
+  ret.ruleEnabledStatus = {
+    enabled: enabledBuckets.find((bucket) => bucket.key === 1)?.doc_count ?? 0,
+    disabled: enabledBuckets.find((bucket) => bucket.key === 0)?.doc_count ?? 0,
+  };
+
+  const mutedBuckets = aggregations.muted.buckets;
+  ret.ruleMutedStatus = {
+    muted: mutedBuckets.find((bucket) => bucket.key === 1)?.doc_count ?? 0,
+    unmuted: mutedBuckets.find((bucket) => bucket.key === 0)?.doc_count ?? 0,
+  };
+
+  ret.ruleSnoozedStatus = {
+    snoozed: aggregations.snoozed?.count?.doc_count ?? 0,
+  };
+
+  const tagsBuckets = aggregations.tags?.buckets || [];
+  ret.ruleTags = tagsBuckets.map((bucket) => bucket.key);
+
+  return ret;
```

Comment:
What does it mean: `ret`?

Reply:
i think it used to mean "return". I've renamed it to "result"